eZ/Publish/Core/REST/Server/Controller/Content.php

```diff
@@ -736,4 +786,66 @@ protected function forward($controller)
 
         return $this->container->get('http_kernel')->handle($subRequest, HttpKernelInterface::SUB_REQUEST);
     }
+
+    /**
+     * Get new main language code traversing through all Versions to look for the oldest one.
+     *
+     * @param \eZ\Publish\API\Repository\Values\Content\VersionInfo $versionInfo
+     * @param string $removedLanguageCode
+     *
+     * @return string
+     *
+     * @throws \eZ\Publish\Core\REST\Client\Exceptions\BadStateException
+     */
+    private function getNewMainLanguageCode(VersionInfo $versionInfo, $removedLanguageCode)
```

Comment:
This method implements automatic update of main Translation of a Content if it is the one we want to delete, following defined rule:

> I think we should just define a rule (and document it for the API), what makes sense to me is the chronological order. Take the oldest language introduced in the content item.

So I implemented it and this led to a very ugly example of spaghetti code and I don't think it's possible to make it better, so I prefer to get rid of it.

Simpler alternative would be to take first language from current version, however the cleanest one would be for REST endpoint not to worry about main Translation at all and throw exception in that case (should be updated via content metadata endpoint prior to deleting Translation). There's a scenario about that, but for similar feature related to Translation removal from all Versions (waiting for PM response in Jira).

// cc @andrerom @bdunogier @dpobel